src/utils/Codec.ts

```diff
@@ -88,6 +55,17 @@ export class Codec {
         return ethers.AbiCoder.defaultAbiCoder().encode([ethersType], [struct]);
     }
 
+    public static encodeWithExplicitType(
```

Comment:
should be also named 'encode' - typescript supports method overloading

Reply:
@MrishoLukamba you didn't change this

Reply:
@daiagi , So am getting linting errors with function overloading and my opinion is in TS we shouldn't do the overload, whats your opinion?

Reply:
@Mrisho
1.  He told you his opinion.

2. "_In my opinion we should not_" 
is not an argument.
This is too basic to be up to opinion. It is a small and pure issue of can do or can't do. 
If you have a issue doing it then post screenshot and backup your claims with link.

3. If you can't overload you could make a function with optional argument that has delegates to two internal functions based on the existence / lack of the argument. 

4. `EncodeWithExplicitType` is way too long  / verbose name.

Kinda OK for internal function. Very not OK for static / public function